examples/can.rs

```diff
@@ -31,10 +32,10 @@ fn main() -> ! {
 
     let _clocks = rcc
         .cfgr
-        .use_hse(32.mhz())
-        .sysclk(32.mhz())
-        .pclk1(16.mhz())
-        .pclk2(16.mhz())
+        .use_hse(Hertz(32_000_000))
```

Comment:
Hm. Now I know why you were going for the result approch before, I've played around a bit to look for alternatives:

```suggestion
        .use_hse(32_000_000_u32.Hz())
```

```suggestion
        .use_hse(32u32.MHz().try_into().unwrap())
```

or even change the corresponding function input type to 

```rust
pub fn use_hse(mut self, freq: Megahertz) -> Self
```

so you can write 

```suggestion
        .use_hse(32u32.MHz())
```

Definitely for `src.rcc.rs` I think the last suggestions should be used. We know that 72.Mhz() is the limit anyways, and this can be safely converted to Hz in u32 bit range.

Though this constrains the resolution to `1.MHz()`, which I'm fine with in the RCC case, because finer resolutions are not really guaranteed anyways. (Maybe I'm wrong though and do overlook something)

For the rest of the modules, I think it is not as easily applicable. I think we should stick to Hertz in these cases. 
In these  cases I slightly prefer my first suggestion in favor of your current approach. But this is not a strong opinion and your current approach is definitely fine!

Reply:
I'll take a look at the `RM` later tonight, just to make sure that won't adversely affect anyone.

Reply:
What are your thoughts on the following:

    /// # Panics
    ///
    /// Panics if conversion from `Megahertz` to `Hertz` produces a 
    /// value greater than `u32::MAX`.
    pub fn use_hse(mut self, freq: Megahertz) -> Self {
        let freq: Hertz = freq.try_into( ).unwrap( );
        self.hse = Some(freq.0);
        self
    }

Doing it like above means minimal internal changes in `rcc`. 
The only change would be to the public API.